v2/tools/generator/internal/conversions/property_conversions.go

```diff
@@ -2021,6 +2052,163 @@ func assignObjectsViaIntermediateObject(
 	}, nil
 }
 
+// assignNoninlineObjectsViaPivotObject will generate a conversion if both properties are TypeNames referencing Object
+// Type definitions, neither property is optional, the types are NOT adjacent in our storage conversion graph, and they
+// are NOT inline with each other. The conversion is implemented by assigning properties to an intermediate pivot
+// instance before assigning those to our actual destination instance.
+//
+// A key difference between this and assignInlineObjectsViaIntermediateObject is that this handles the case where
+// the types are on different branches of the conversion graph, necessitating discovery of the closest shared type
+// to use as a pivot. Using this kind of pivot requires reversing the direction of the second half of the conversion!
+//
+// For ConvertFrom the generated code will be:
+//
+// var <pivot> <pivotType>
+// err := <pivot>.AssignPropertiesFrom(<source>)
+//
+//	if err != nil {
+//	   return errors.Wrap(err, "while calling <pivot>.AssignPropertiesFrom(<source>)")
+//	}
+//
+// var <otherlocal> <destinationType>
+// err := <pviot>.AssignPropertiesTo(<otherLocakl>)
+//
+//	if err != nil {
+//	   return errors.Wrap(err, "while calling <otherlocal>.AssignPropertiesFrom(<local>)")
+//	}
+//
+// Note the actual steps are generated by nested conversions; this handler works by finding the two conversions needed
+// given our intermediate type and chaining them together.
+func assignNoninlineObjectsViaPivotObject(
```

Comment:
minor, should this be:
```suggestion
func assignNonInlineObjectsViaPivotObject(
```

Since I think `non-inline` would be cased like that?

Reply:
Changed.